corehq/form_processor/models.py

```diff
@@ -88,4 +92,132 @@ def read_content(self):
+    case_type = models.CharField(max_length=255)
+
+    owner_id = models.CharField(max_length=255)
+
+    opened_on = models.DateTimeField(null=False)
+    opened_by = models.CharField(max_length=255, null=False)
+
+    modified_on = models.DateTimeField(null=False)
+    server_modified_on = models.DateTimeField(null=False)
+    modified_by = models.CharField(max_length=255)
+
+    closed = models.BooleanField(default=False, null=False)
+    closed_on = models.DateTimeField(null=True)
+    closed_by = models.CharField(max_length=255, null=False)
+
+    deleted = models.BooleanField(default=False, null=False)
+
+    external_id = models.CharField(max_length=255)
+
+    case_json = JSONField(lazy=True)
+    attachments_json = JSONField(lazy=True)
+
+    @property
+    def case_id(self):
+        return self.case_uuid
+
+    def hard_delete(self):
+        self.delete()
+
+    def soft_delete(self):
+        self.deleted = True
+        self.save()
+
+    def is_deleted(self):
+        return self.deleted
+
+    def get_attachment(self, attachment_name):
+        assert attachment_name in self.attachments_json
+        with open(self._get_attachment_path(attachment_name), 'r+') as f:
+            content = f.read()
+        return content
+
+    def _get_attachment_path(self, attachment_name):
+        attachment_id = '{}_{}'.format(self.case_uuid, attachment_name)
+        if getattr(settings, 'IS_TRAVIS', False):
+            return os.path.join('/home/travis/', attachment_id)
+        return os.path.join('/tmp/', attachment_id)
+
+    @classmethod
+    def get(cls, case_id):
+        return CommCareCaseSQL.objects.get(case_uuid=case_id)
+
+    @classmethod
+    def get_cases(cls, ids):
+        return CommCareCaseSQL.objects.filter(case_uuid__in=list(ids))
+
+    @classmethod
+    def get_case_xform_ids(cls, case_id):
+        return CaseForms.objects.filter(case_uuid=case_id)
+
+    @classmethod
+    def get_obj_id(cls, obj):
+        return obj.case_uuid
+
+    @classmethod
+    def get_obj_by_id(cls, _id):
+        return CommCareCaseSQL.objects.get(_id)
+
+    def __unicode__(self):
+        return (
+            "CommCareCase("
+            "case_id='{c.case_uuid}', "
+            "domain='{c.domain}', "
+            "closed={c.closed}, "
+            "owner_id='{c.owner_id}', "
+            "server_modified_on='{c.server_modified_on}')"
+        ).format(c=self)
+
+    class Meta:
+        # TODO SK 2015-11-05: verify that these are the indexes we want
+        # also consider partial indexes
+        index_together = [
+            ["domain", "owner_id"],
+            ["domain", "closed", "server_modified_on"],
+        ]
+
+
+class CommCareCaseIndexSQL(models.Model):
+    CHILD = 0
+    EXTENSION = 1
+    RELATIONSHIP_CHOICES = (
+        (CHILD, 'child'),
+        (EXTENSION, 'extension'),
+    )
+
+    case = models.ForeignKey('CommCareCaseSQL', to_field='case_uuid', db_column='case_uuid', db_index=True)
```

Comment:
also very minor, i don't think you need `db_column` since it's the same as the `to_field`